File: FAÇADE (1)/proxy01.js
//Objeto Real1
class ServicoReal{
  executar(){
    console.log(`Executando Servico Real`);
  }
}


//Proxy
class ProxyServico{
  constructor(){
    this.servicoReal = new ServicoReal();
  }

  executar(){
    console.log(`Antes de executar do servico`);
    this.servicoReal.executar();
    console.log(`Depois de executar de servico`);
  }
}


//Cliente - USO DO PROXY
const proxy = new ProxyServico();
proxy.executar();

File: FAÇADE (1)/proxy01.py
#Objeto Real 1
class ServicoReal:
    def executar(self):
        print("Método executar do Serviço Real.")


#Proxy
class ProxyServico:
    def executar(self, servicoReal):
        print("Antes de executar do serviço")
        servicoReal.executar()
        print("Depois de executar do serviço")


#Cliente - USO DO PROXY
servico_real = ServicoReal()
proxy = ProxyServico()
proxy.executar(servico_real)
